ComplianceSuite/Tests/RegoComplianceTests/RegoComplianceTests.swift

```diff
@@ -121,8 +123,113 @@ struct ComplianceTests {
         }
         return
     }
+    
+    private static func addTestBuiltins(to registry: BuiltinRegistry) -> BuiltinRegistry {
+        guard !registry.hasBuiltin("test.sleep") else {
+            return registry
+        }
+
+        let newRegistry = BuiltinRegistry(
+            builtins: registry.builtins.merging(
+                ["test.sleep" : TestBuiltins.test_sleep],
+                uniquingKeysWith: { (current, _) in current })
+        )
+        return newRegistry
+    }
 }
 
 extension ComplianceTesting.IRTestCase: CustomTestStringConvertible {
     public var testDescription: String { "\(description)" }
 }
+
+struct TestBuiltins {
+    static func test_sleep(ctx: BuiltinContext, args: [AST.RegoValue]) async throws -> AST.RegoValue {
+        guard args.count == 1 else {
+            throw BuiltinError.argumentCountMismatch(got: args.count, want: 1)
+        }
+
+        guard case .string(let x) = args[0] else {
+            throw BuiltinError.argumentTypeMismatch(arg: "x", got: args[0].typeName, want: "string")
+        }
+
+        guard let timeDuration = try? parseDurationNanoseconds(x), timeDuration > 0 else {
+            return .null
+        }
+
+        try await Task.sleep(nanoseconds: UInt64(timeDuration))
+
+        return .null
+    }
+
+    enum DurationParseError: Error {
+        case invalidFormat
+    }
+
+    /// Parses a duration string and returns the equivalent duration in nanoseconds.
+    ///
+    /// - Parameters:
+    ///   - duration: A string representing a duration with parts in the format of numeric values followed by a unit (e.g., "10m", "1s", "10ms").
+    ///
+    /// - Returns: The duration represented as an `Int64` in nanoseconds.
+    ///
+    /// - Throws: `DurationParseError.invalidFormat` if the input string doesn't conform to the expected format.
+    ///
+    /// - Note: Supports units such as nanoseconds (ns), microseconds (us, µs),
+    ///        milliseconds (ms), seconds (s), minutes (m), and hours (h).
+    static func parseDurationNanoseconds(_ duration: String) throws -> Int64 {
+        var isNegative = false
+        var durationValue = duration
+
+        // Check for negative sign at the start
+        if duration.hasPrefix("-") {
+            isNegative = true
+            durationValue = String(duration.dropFirst())
+        }
+
+        let pattern = "(\\d+)(ns|us|µs|ms|s|m|h)"
+        let regex = try! NSRegularExpression(pattern: pattern, options: [])
```

Comment:
⛏️ we can use Swift-native regular expressions instead of the older APIs:

```
let regex = /(?<value>\d+)(?<unit>ns|us|µs|ms|s|m|h)/
let results = durationValue.matches(of: regex)

var totalNanoseconds: Int64 = 0

// Keep matching the string accumulating nanoseconds
for match in results {
    let valuePart = match.value
    let unitPart = match.unit
    ...
}
```

Reply:
Should be fixed in the next commit. Thanks!